src/pdiagmat.jl

```diff
@@ -74,63 +74,35 @@ LinearAlgebra.sqrt(a::PDiagMat) = PDiagMat(map(sqrt, a.diag))
 
 ### whiten and unwhiten
 
-function whiten!(r::StridedVector, a::PDiagMat, x::StridedVector)
-    n = dim(a)
-    @check_argdims length(r) == length(x) == n
-    v = a.diag
-    for i = 1:n
-        r[i] = x[i] / sqrt(v[i])
-    end
-    return r
-end
-
-function unwhiten!(r::StridedVector, a::PDiagMat, x::StridedVector)
-    n = dim(a)
-    @check_argdims length(r) == length(x) == n
-    v = a.diag
-    for i = 1:n
-        r[i] = x[i] * sqrt(v[i])
-    end
-    return r
-end
-
-function whiten!(r::StridedMatrix, a::PDiagMat, x::StridedMatrix)
-    r .= x ./ sqrt.(a.diag)
-    return r
-end
-
-function unwhiten!(r::StridedMatrix, a::PDiagMat, x::StridedMatrix)
-    r .= x .* sqrt.(a.diag)
-    return r
-end
-
+whiten!(r::AbstractVecOrMat, a::PDiagMat, x::AbstractVecOrMat) = r .= x ./ sqrt.(a.diag)
+unwhiten!(r::AbstractVecOrMat, a::PDiagMat, x::AbstractVecOrMat) = r .= x .* sqrt.(a.diag)
 
 ### quadratic forms
 
 quad(a::PDiagMat, x::AbstractVector) = wsumsq(a.diag, x)
 invquad(a::PDiagMat, x::AbstractVector) = invwsumsq(a.diag, x)
 
-function quad!(r::AbstractArray, a::PDiagMat, x::StridedMatrix)
+function quad!(r::AbstractArray, a::PDiagMat, x::AbstractMatrix)
     m, n = size(x)
     ad = a.diag
     @check_argdims m == length(ad) && length(r) == n
-    @inbounds for j = 1:n
+    @inbounds for j in axes(x, 2)
         s = zero(promote_type(eltype(ad), eltype(x)))
-        for i in 1:m
+        for i in axes(x, 1)
```

Comment:
Same here, i is possibly invalid for indexing ad.
```suggestion
        for (adi, i) in zip(ad, axes(x, 1))
```

Reply:
fixed check (just noting here so will push shortly)